src/ampycloud/data.py

```diff
@@ -1036,6 +1076,9 @@ def metar_msg(self, which: str = 'layers') -> str:
 
         # Here, deal with the situations when all clouds are above the MSA
         if len(msg) == 0:
-            return 'NCD'
+            sligrolay_in_buffer = sligrolay['significant'] * (sligrolay['alt_base'] >= msa_val)
+            if sligrolay_in_buffer.any():  # clouds within the MSA buffer zone
+                return 'NSC'
+            return self._ncd_or_nsc()
```

Comment:
To check if I understood correctly: 

- line 1080 checks whether there are any cloud layer bases above the MSA (after running ampycloud)
- line 1082 checks whether there are any ceilometer hits above the MSA (before running ampycloud)

I am a bit confused and additional comments into the code would help.

Reply:
added a comment.